src/api/SearchApi.js

```diff
@@ -74,39 +65,53 @@ class RequestSerializer {
   };
 }
 
-/** Default backend response serializer */
-class ResponseSerializer {
-  _serializeAggregation(bucket) {
-    const aggregation = {
-      key: bucket.key,
-      total: bucket.doc_count,
-      hasNestedField: false,
-    };
+const extractFromPath = (aggregation, pathToExtract) => {
+  return pathToExtract.reduce((obj, key) =>
+    (obj && obj[key] !== 'undefined') ? obj[key] : undefined, aggregation);
+}
+
+const createAggregation = (parentKeyName, buckets) => {
```

Comment:
What do you think if you encapsulate both functions inside `ResponseSerializer` class?